qtdialogs/DlgSetupManager.py

```diff
@@ -0,0 +1,1197 @@
+         for idx in range(len(wltList)):
+            entry = wltList[idx]
+            # Derive display values
+            walletPath = getattr(entry, 'path', '')
+            fileName = os.path.basename(walletPath) if walletPath else ''
+            walletId = getattr(entry, 'walletId', '')
+            loadState = getattr(entry, 'state', 0)
+
+            # Create row
+            item = QtWidgets.QTreeWidgetItem()
+            item.setText(0, walletId if walletId else fileName)
+            item.setText(1, getattr(entry, 'title', walletId or fileName))
+
+            # Determine state and action
+            # WalletManagerReply.WalletLoadState: 1=legacy, 3=encrypted, 4=ready
+            # loadState is a capnp enum; rely on underlying numeric value via string compare
+            loadStateStr = str(loadState)
+            isLegacy = (loadStateStr.endswith('.legacy')) or (not bool(walletId))
+            if isLegacy:
+               stateText = 'Legacy'
+            elif loadStateStr.endswith('.encrypted'):
+               stateText = 'Encrypted (locked)'
+            else:
+               stateText = 'Ready'
+            item.setText(2, stateText)
+
+            # Action widget
+            actionWidget = QtWidgets.QWidget()
+            actionLayout = QtWidgets.QHBoxLayout(actionWidget)
+            actionLayout.setContentsMargins(0, 0, 0, 0)
+            actionBtn = QtWidgets.QPushButton()
+            actionBtn.setStyleSheet(UI_STYLE_BUTTON_STANDARD)
+
+            fullPath = walletPath if walletPath else os.path.join(self.armoryDataDirEdit.text(), fileName)
+
+            if isLegacy:
+               actionBtn.setText(self.tr('Migrate'))
+               actionBtn.clicked.connect(lambda _, p=fullPath: self._migrateWallet(p))
+            elif loadStateStr.endswith('.encrypted'):
+               actionBtn.setText(self.tr('Decrypt'))
+               actionBtn.clicked.connect(lambda _, wid=walletId: self._unlockWallet(wid))
+            else:
+               actionBtn.setText(self.tr('Load'))
+               actionBtn.clicked.connect(lambda _, wid=walletId: self._loadWallet(wid))
+
+            actionLayout.addWidget(actionBtn)
+            self.walletList.addTopLevelItem(item)
+            self.walletList.setItemWidget(item, 3, actionWidget)
+
+      except Exception as e:
+         LOGEXCEPT('Error loading wallet list: %s', str(e))
+
+   def _migrateWallet(self, walletPath):
+      try:
+         # DlgMigrateWallet(promptId, wltID, verbose, parent=None, main=None)
+         DlgMigrateWallet(None, walletPath, self.tr('Migrate'), self, self).exec_()
+         TheBridge.wltManager.loadWallets()
+         self.loadWalletList()
+      except Exception as e:
+         QtWidgets.QMessageBox.warning(self, self.tr('Migration Failed'), str(e))
+
+   def _unlockWallet(self, walletId):
+      try:
+         # Control header unlock requires a path; list again to find the entry
+         wltList = TheBridge.wltManager.listWallets()
```

Comment:
dont ask BE for wallets when py side store has them. this will fuck a bunch of things when multiple wallets are present